HybrasylXML/XSD/hybrasylExtensions.cs

```diff
@@ -222,20 +222,19 @@ public byte GetMaxLevelByClass(Class castableClass)
         public bool TryGetMotion(Class castClass, out Motion motion)
         {
             motion = null;
-            try
-            {
-                motion = Effects.Animations.OnCast.Player.SingleOrDefault(x => x.Class.Contains(castClass));
-            }
-            catch (InvalidOperationException)
-            {
-                motion = Effects.Animations.OnCast.Player.FirstOrDefault(x => x.Class.Contains(castClass));
-            }
-            catch (NullReferenceException)
-            {
+
+            if (Effects?.Animations?.OnCast?.Player == null) return false;
+
+            var m = Effects.Animations.OnCast.Player.Where(e => e.Class.Contains(castClass));
+            if (m.Count() == 0)
+                m = Effects.Animations.OnCast.Player.Where(e => e.Class.Count == 0);
+
+            if (m.Count() == 0)
```

Comment:
This should remain singleordefault, as Where will return an enumerable, and single will return the actual object. SingleOrDefault will return properly. Default should be defined on the animation.